app/views/users/index.html.erb

```diff
@@ -2,15 +2,40 @@
 
 <table class="table">
   <thead>
-    <th>Name</th>
-    <th>Email</th>
-    <th>Last sign in</th>
+    <th><%= User.human_attribute_name('name') %></th>
+    <th><%= User.human_attribute_name('email') %></th>
+    <% if current_user.admin? %>
+      <th><%= User.human_attribute_name('role') %></th>
+    <% end %>
+    <th><%= User.human_attribute_name('last_sign_in_at') %></th>
+    <th></th>
   </thead>
   <tbody>
     <% @users.each do |user| %>
       <tr>
         <td><%= link_to user.name, user_path(user) %></td>
         <td><%= user.email %></td>
+        <% if current_user.admin? %>
+          <td>
+            <div class="btn-group" role="group">
+              <% User.roles.map(&:first).each do |role| %>
+                <% if last_admin?(user) %>
```

Comment:
I think you can write `last_admin?(user) && current_user?(user)` or define a new function to avoid having `current_user?` inside the `last_admin` function.